dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseAnalyticalObject.java

```diff
@@ -452,7 +452,7 @@ protected DimensionalObject getDimensionalObject(
 
```

Comment:
> Had to base the PR on fix/visualizations-relative-periods-import-error because it updates getRelatives() with code needed for this fix.

But the current base is fix/ocg-40, that will close the PR when merged. Do we want "fix/visualizations-relative-periods-import-error" as base? 

This PR fixes `getDimensionalObject`, not sure if we need to start at fix/visualizations-relative-periods-import-error, though, but I don't have the full picture.

Reply:
`fix/visualizations-relative-periods-import-error` changes:

```
  public RelativePeriods getRelatives() {
    return relatives;
  }
```
To:
```
  public RelativePeriods getRelatives() {
    if (relatives == null) {
      List<RelativePeriodEnum> enums = new ArrayList<>();

      if (rawRelativePeriods != null) {
        for (String relativePeriod : rawRelativePeriods) {
          if (RelativePeriodEnum.contains(relativePeriod)) {
            enums.add(RelativePeriodEnum.valueOf(relativePeriod));
          }
        }
      }

      return new RelativePeriods().setRelativePeriodsFromEnums(enums);
    }

    return relatives;
  }
  ````
I started from `fix/visualizations-relative-periods-import-error` because it looks like it doesn't fixes anything otherwise.

Reply:
I saw the commit being merged into fix/ocg-40 so i assumed the PR was already closed. Also out of habit to have a specific PR for a task.